### PR TITLE
Week 11, Happy Thoughts - Therese Hagelin

### PR DESCRIPTION
**Pull request by ThereseHag**
Link to my deployed project:

https://therese-happy-thoughts.netlify.app/

My reflections on how this week's project turned out:

This weeks project has been a joy. It has worked out really well. Working with API's is something I find really fun, useful and interesting. 

Things I'd like to have clarified or explained in more detail:

I still haven't tried to add the character counter since I didn't have time for it at the end of the week, but in general, I would love to learn more about this kind of counters and conditional things.

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,83 @@
...
+  }
+
+  const handleMessageNewChange = (event) => {
+    setMessageNew(event.target.value);
+  }
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    event.target.reset();
+
```
Comment (honkoan) [inline code/src/App.js:31]: This comment is probably more for me but how on earth I didn't manage to get this one line to work in my code. 😆

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,83 @@
...
+     />
+     <MessageList 
+     messageList = {messageList}
+     handleLikesIncrease = {handleLikesIncrease}
+     />
+     
     </div>
+    
   )
 }
```
Comment (honkoan) [inline code/src/App.js:83]: Your App.js file is well structured and easy to understand. It is a pleasure to read your code.

File: code/src/components/MessageList.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+           key = {message._id}
+           message = {message}
+           onLikesIncrease = {handleLikesIncrease}  
+           /> 
+          ) )}
+          </>
+    )
+}
+
+export default MessageList;
```
Comment (honkoan) [inline code/src/components/MessageList.js:19]: Your all components are clearly structured and easy to understand. Good job! 🥳

File: code/src/index.css [hunk 4]
```diff
@@ -11,3 +11,96 @@ code {
...
+  flex-direction: row;
+  padding: 10px;
+  margin: 10px;
+}
+
+.heart-count {
+  margin: 10px;
+  padding: 10px;  
+}
+
```
Comment (honkoan) [inline code/src/index.css:106]: You really know how to name the classes! Your CSS-file is easy to understand when your classnames are descriptive.

File: code/src/components/MessageElement.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+              <h4>{message.message}</h4>
+              <div className="button-wrapper">
+              <button className="heart-button" onClick={() => onLikesIncrease(message._id)}>
+              <span className={message.hearts > 0 ? 'heart-button-active' : 'heart-button-not-active'} role='img' aria-label='heart icon'>❤</span>
+              </button>
+              <p className="heart-count">x {message.hearts}</p>
+              
+              <p className="date">- {moment(message.createdAt).fromNow()}</p>
+              </div>
+            </div>
```
Comment (honkoan) [inline code/src/components/MessageElement.js:16]: Here you could check the indentation and remove the empty line. 🙂

**Review by honkoan — APPROVED** [review]
Great job with this project Therese! 🥳 

Your App is working and made me smile even though I have also made the same project myself. You wrote that this project was a joy. It was also a joy to read your code. 🙂 
Your code is well structured and clean and easy to understand.

The app fulfills all the blue level requirements and also one red level requirement. Very good job, Therese! 🙌🏻 

There is not many things that I can comment on your project because it is well done (and the code looks pretty much the same as my code 🤣). But if you want to pay attention to something, you could check the indentation and spacing in some components. I added a comment about the indentation to "MessageElement"-component, but you could check the indentation in other components too, if you want. 

In summary: Very good project!

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,83 @@
...
+
+  useEffect(() => {
+    fetchMessageList();
+  }, []);
+
+  const fetchMessageList = () => {
+    fetch(API_URL)
+    .then(res => res.json())
+    .then(messages => setMessageList(messages))
+    .catch(err => console.error(err));
```
Comment (KarolCislowski) [inline code/src/App.js:21]: console outputs can't be present in the production code.

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Hej!
Great job, but please clean your code from all console outputs 🧹

**Review by KarolCislowski — APPROVED** [review]
😸
